src/main/java/com/urbancode/jenkins/plugins/ucdeploy/VersionHelper.java

```diff
@@ -295,6 +295,8 @@ private void setComponentVersionProperties(
                 propSheetDef = compClient.getComponentVersionPropSheetDef(component);
                 propSheetDefId = (String) propSheetDef.get("id");
                 propSheetDefPath = (String) propSheetDef.get("path");
+                propSheetDefPath = propSheetDefPath.replaceAll("/", "&");
```

Comment:
Nick,

Here is the output from the test.groovy run on the Jenkins Redhat linux 7.2 slave using groovy 1.8.8 from the udeploy agent install. This is the same server that the Jenkins pipeline job is running on and these results are consistent with the Jenkins job. We have the same results on other slaves and I get the same behavior in Chrome and IE web browsers.
We are on CloudBees Jenkins Enterprise 2.32.2.7-rolling 

[srvctgdoadm@vc2cmmkb182811n ~]$ /jenkins/ibm-ucd-qa/agent/opt/groovy-1.8.8/bin/groovy -cp "./uc-uDeployRestClient-1.0-SNAPSHOT.jar" test.groovy
java.io.IOException: 404 Not Found
Resource not found:/property/propSheetDef/components/61b5b5ca-3427-4238-bc31-3d3c22ca0b71/environmentPropSheetDef.-1/propDefs
        at com.urbancode.ud.client.UDRestClient.invokeMethod(UDRestClient.java:204)
        at com.urbancode.ud.client.PropertyClient.getPropSheetDefPropDefs(PropertyClient.java:94)
        at com.urbancode.ud.client.PropertyClient$getPropSheetDefPropDefs.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:42)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at test.run(test.groovy:16)
        at groovy.lang.GroovyShell.runScriptOrMainOrTestOrRunnable(GroovyShell.java:266)
        at groovy.lang.GroovyShell.run(GroovyShell.java:229)
        at groovy.lang.GroovyShell.run(GroovyShell.java:159)
        at groovy.ui.GroovyMain.processOnce(GroovyMain.java:550)
        at groovy.ui.GroovyMain.run(GroovyMain.java:337)
        at groovy.ui.GroovyMain.process(GroovyMain.java:323)
        at groovy.ui.GroovyMain.processArgs(GroovyMain.java:120)
        at groovy.ui.GroovyMain.main(GroovyMain.java:100)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.groovy.tools.GroovyStarter.rootLoader(GroovyStarter.java:108)
        at org.codehaus.groovy.tools.GroovyStarter.main(GroovyStarter.java:130)
propSheetDefPathAmper:[{"id":"a904b155-19b5-4e7c-b689-8df73a0d1acd","name":"REMOTE_DEPLOY_DIR","label":"REMOTE_DEPLOY_DIR","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":true}]

java.io.IOException: 404 Not Found
Resource not found:/property/propSheetDef/components/61b5b5ca-3427-4238-bc31-3d3c22ca0b71/versionPropSheetDef.-1/propDefs
        at com.urbancode.ud.client.UDRestClient.invokeMethod(UDRestClient.java:204)
        at com.urbancode.ud.client.PropertyClient.getPropSheetDefPropDefs(PropertyClient.java:94)
        at com.urbancode.ud.client.PropertyClient$getPropSheetDefPropDefs.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:42)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at test.run(test.groovy:34)
        at groovy.lang.GroovyShell.runScriptOrMainOrTestOrRunnable(GroovyShell.java:266)
        at groovy.lang.GroovyShell.run(GroovyShell.java:229)
        at groovy.lang.GroovyShell.run(GroovyShell.java:159)
        at groovy.ui.GroovyMain.processOnce(GroovyMain.java:550)
        at groovy.ui.GroovyMain.run(GroovyMain.java:337)
        at groovy.ui.GroovyMain.process(GroovyMain.java:323)
        at groovy.ui.GroovyMain.processArgs(GroovyMain.java:120)
        at groovy.ui.GroovyMain.main(GroovyMain.java:100)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.groovy.tools.GroovyStarter.rootLoader(GroovyStarter.java:108)
        at org.codehaus.groovy.tools.GroovyStarter.main(GroovyStarter.java:130)
propSheetDefPathAmper:[{"id":"3f61b07a-0745-4713-8775-0b6207d1d954","name":"buildSource","label":"buildSource","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"66cc01e6-154e-4254-b32e-6214c9f88e59","name":"buildTarget","label":"buildTarget","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"d1c590f6-0e8a-4d3c-bab2-016ea6e47a2b","name":"buildName","label":"buildName","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"37b1e635-90c2-4e89-b69c-0f2606baf345","name":"buildNumber","label":"buildNumber","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"338b0be0-7d29-4b74-9b32-3c79615340a2","name":"buildStatus","label":"buildStatus","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"8fecb6e1-486a-4c56-96be-097a8ecb29d5","name":"buildComment","label":"buildComment","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false},{"id":"4e275ac9-416c-4b81-803a-54523be0d51f","name":"buildProps","label":"buildProps","pattern":"","type":"TEXT","value":"","required":false,"description":"","placeholder":"","inherited":false}]